src/core_plugins/getting_started/public/lib/add_setup_work.js

```diff
@@ -68,14 +69,14 @@ uiRoutes
         });
         notify.error('Please create a new index pattern');
         kbnUrl.change(CREATE_INDEX_PATTERN_ROUTE);
-        return;
+        return Promise.halt(); //prevent route resolve
```

Comment:
It might be good to explain for the benefit of our future selves *why* this is necessary via a comment in this file somewhere. Since this is being done in a couple of places maybe we can extract a function here that changes the route and return a halted promise?